public/push-messaging-sw.js

```diff
@@ -0,0 +1,30 @@
+/* global importScripts firebase FCM_SENDER_ID */
+// Give the service worker access to Firebase Messaging.
+// Note that you can only use Firebase Messaging here, other Firebase libraries
+// are not available in the service worker.
+
+importScripts('/lib/firebase/firebase-app.js');
+importScripts('/lib/firebase/firebase-messaging.js');
+importScripts('/config/sw.js');
+
+firebase.initializeApp({
+  'messagingSenderId': FCM_SENDER_ID
+});
+
+var messaging = firebase.messaging();
+
+messaging.setBackgroundMessageHandler(function(payload) {
+  // not actually used, but without it here firefox does not receive messages...
+  console.log('received payload', payload);
```

Comment:
No angular here, the service worker is a standalone script and they run with a limited environment.

https://github.com/firebase/quickstart-js/issues/71 has a discussion about it - the gist is that by sending a notification payload (as opposed to data) you are specifying it should be a regular notification, and not be overridable.

my suspicion about why it didn't work without that in firefox is that without defining _some_ kind of listener it ends. it would be useful to verify it properly though...